solr/solr_chart/main.tf

```diff
@@ -0,0 +1,29 @@
+data "digitalocean_kubernetes_cluster" "dev" {
+  name = "dev-k8s-treetracker"
+}
+
+provider "helm" {
+  kubernetes {
+    host  = data.digitalocean_kubernetes_cluster.dev.endpoint
+    token = data.digitalocean_kubernetes_cluster.dev.kube_config[0].token
+    cluster_ca_certificate = base64decode(
+      data.digitalocean_kubernetes_cluster.dev.kube_config[0].cluster_ca_certificate
+    )
+  }
+}
+
+resource "helm_release" "solr" {
+  name             = "solr-operator"
+  repository       = "https://solr.apache.org/charts"
+  chart            = "solr-operator"
+  version          = "0.7.0"
+  namespace        = "solr"
+  create_namespace = true
+
+
+  values = [
+    "${file("${path.module}/values.yaml")}",
+    var.values_file
+  ]
+
+}
```

Comment:
@mckornfield I'm trying to understand what you are doing, let me know if I was wrong:
1. We are using Terraform's helm provider to install Solr to K8S
2. What are the benefits of using Terraform  rather then use ansible?
3. What's the things need to save to Terraform states?
4. I can not find the code to set the namespace for the Solr?

Reply:
1. Yep that's right
2. To be honest, it's about the same, except that there's a state reference somewhere. I've recently come to use terraform for cluster installations that are more permanent, and I think I like it more than ansible, as far as being able to "show" what the current state is, and detect drift
3. That script I added basically pulls keys from a locally filled out `~/.netrc` to query the appropriate keys
4. https://github.com/Greenstand/treetracker-infrastructure/pull/165/files#diff-24ac15345fd7e132af56f8f88808b1d8196f8ad1500cbbd46ce36851b51cfca3R20 this line sets the namespace

Reply:
Thank you for you great answer!